plugins/xsettings/msd-xsettings-manager.c

```diff
@@ -534,8 +534,11 @@ scale_change_workarounds (MateXSettingsManager *manager, int new_scale)
                 gsettings = g_hash_table_lookup (manager->priv->gsettings, INTERFACE_SCHEMA);
                 /* If enabled, set env variables to properly scale QT applications */
                 if (g_settings_get_boolean (gsettings, SCALING_FACTOR_QT_KEY)) {
-                        if (!update_user_env_variable ("QT_AUTO_SCREEN_SCALE_FACTOR", "0", &error)) {
-                                g_warning ("There was a problem when setting QT_AUTO_SCREEN_SCALE_FACTOR=0: %s", error->message);
+                        char dpibuf[G_ASCII_DTOSTR_BUF_SIZE];
```

Comment:
You don't need 29 + 10 characters for printing a 64 bit integer, you need 21 chars log(2^64)+1 (\0 NULL terminated string).
https://developer.gnome.org/glib/stable/glib-String-Utility-Functions.html#G-ASCII-DTOSTR-BUF-SIZE:CAPS

Reply:
Technically yes, but it's not worth introducing another constant for this (I don't see anything like this in glib specifically for integers). I simply reused the same size that was used for `dpibuf` in `xft_settings_set_xresources`. This is a shortlived stack allocation (only within this function), it just does not matter. I would say let's keep this as is.

Reply:
```diff
diff --git a/plugins/xsettings/msd-xsettings-manager.c b/plugins/xsettings/msd-xsettings-manager.c
index 861aa6a..0add60f 100644
--- a/plugins/xsettings/msd-xsettings-manager.c
+++ b/plugins/xsettings/msd-xsettings-manager.c
@@ -534,9 +534,9 @@ scale_change_workarounds (MateXSettingsManager *manager, int new_scale, int unsc
                 gsettings = g_hash_table_lookup (manager->priv->gsettings, INTERFACE_SCHEMA);
                 /* If enabled, set env variables to properly scale QT applications */
                 if (g_settings_get_boolean (gsettings, SCALING_FACTOR_QT_KEY)) {
-                        char dpibuf[G_ASCII_DTOSTR_BUF_SIZE];
-                        g_snprintf (dpibuf, sizeof (dpibuf), "%d", (int) (unscaled_dpi / 1024.0 + 0.5));
+                        char *dpibuf;
 
+                        dpibuf = g_strdup_printf ("%d", (int) (unscaled_dpi / 1024.0 + 0.5));
                         if (!update_user_env_variable ("QT_FONT_DPI", dpibuf, &error)) {
                                 g_warning ("There was a problem when setting QT_FONT_DPI=%s: %s", dpibuf, error->message);
                                 g_clear_error (&error);
@@ -545,6 +545,7 @@ scale_change_workarounds (MateXSettingsManager *manager, int new_scale, int unsc
                                 g_warning ("There was a problem when setting QT_SCALE_FACTOR=%d: %s", new_scale, error->message);
                                 g_clear_error (&error);
                         }
+                        g_free (dpibuf);
                 }
         } else {
                 /* Restart marco */
```

Reply:
This is arguably worse, since this introduces a heap allocation/deallocation, which is much less efficient than stack allocation (which is literally just incrementing/decrementing the stack pointer register). It's more likely that you lose more memory on heap bookkeeping structures than you have saved. And yet it also does not matter since this memory is only allocated for the scope of this function either way.

My recommendation is still to keep the code exactly as written already.

Reply:
do you use `g_ascii_dtostr` now?

Reply:
No. If you really really care about this, we could do something like `char dpibuf[32];`, given I've seen some prior art in other files here in form of `char buffer[256];`.

I used `G_ASCII_DTOSTR_BUF_SIZE` even though it's for doubles, because it's also enough for ints, so might as well not introduce another constant or use a magic number.

This is seriously the most trivial irrelevant detail.

Reply:
Previously the [g_ascii_dtostr](https://developer.gnome.org/glib/stable/glib-String-Utility-Functions.html#g-ascii-dtostr) function was used to set an environment variable with '.' as decimal part separator, in this case it makes sense to use a print buffer.

I also recommend that you do not use the same buffer to set the cursor.size and Xft.dpi environment variables in xft_settings_set_xresources method.

Reply:
Honestly, we're wasting time on this. The patch is most definitely fine as is.

If you think it's worth continuing the time wastage, just reformat it exactly as you like and we'll go with that.
I only care that the values are set correctly, this stuff is irrelevant.

Reply:
I wrote #369 once it was merged...